heron/schedulers/src/java/org/apache/heron/scheduler/kubernetes/V1Controller.java

```diff
@@ -1101,80 +1104,212 @@ protected V1ConfigMap getConfigMap(String configMapName) {
+          .endEmptyDir()
+          .build();
+
+      for (Map.Entry<KubernetesConstants.VolumeConfigKeys, String> config
+          : configs.getValue().entrySet()) {
+        switch(config.getKey()) {
+          case medium:
+            volume.getEmptyDir().medium(config.getValue());
+            break;
+          case sizeLimit:
+            volume.getEmptyDir().sizeLimit(new Quantity(config.getValue()));
+            break;
+          default:
+            break;
+        }
       }
-      mountList.add(volumeMount.build());
+      volumes.add(volume);
+      volumeMounts.add(createVolumeMountsCLI(volumeName, configs.getValue()));
     }
-    return new Pair<>(volumeList, mountList);
   }
 
   /**
-   * Makes a call to generate <code>Volumes</code> and <code>Volume Mounts</code> and then inserts them.
+   * Generates the <code>Volume</code>s and <code>Volume Mounts</code> for <code>Host Path</code>s to be
+   * placed in the <code>Executor</code> and <code>Manager</code> from options on the CLI.
+   * @param mapOfOpts Mapping of <code>Volume</code> option <code>key-value</code> configuration pairs.
+   * @param volumes A list of <code>Volume</code> to append to.
+   * @param volumeMounts A list of <code>Volume Mounts</code> to append to.
+   */
+  @VisibleForTesting
+  protected void createVolumeAndMountsHostPathCLI(
+      Map<String, Map<KubernetesConstants.VolumeConfigKeys, String>> mapOfOpts,
+      List<V1Volume> volumes, List<V1VolumeMount> volumeMounts) {
+    for (Map.Entry<String, Map<KubernetesConstants.VolumeConfigKeys, String>> configs
+        : mapOfOpts.entrySet()) {
+      final String volumeName = configs.getKey();
+      final V1Volume volume = new V1VolumeBuilder()
+          .withName(volumeName)
+          .withNewHostPath()
+          .endHostPath()
+          .build();
+
+      for (Map.Entry<KubernetesConstants.VolumeConfigKeys, String> config
+          : configs.getValue().entrySet()) {
+        switch(config.getKey()) {
+          case type:
+            volume.getHostPath().setType(config.getValue());
+            break;
+          case pathOnHost:
+            volume.getHostPath().setPath(config.getValue());
+            break;
+          default:
+            break;
+        }
+      }
+      volumes.add(volume);
+      volumeMounts.add(createVolumeMountsCLI(volumeName, configs.getValue()));
+    }
+  }
+
+  /**
+   * Generates the <code>Volume</code>s and <code>Volume Mounts</code> for <code>NFS</code>s to be
+   * placed in the <code>Executor</code> and <code>Manager</code> from options on the CLI.
+   * @param mapOfOpts Mapping of <code>Volume</code> option <code>key-value</code> configuration pairs.
+   * @param volumes A list of <code>Volume</code> to append to.
+   * @param volumeMounts A list of <code>Volume Mounts</code> to append to.
+   */
+  @VisibleForTesting
+  protected void createVolumeAndMountsNFSCLI(
+      Map<String, Map<KubernetesConstants.VolumeConfigKeys, String>> mapOfOpts,
```

Comment:
Would it be worthwhile to mark this parameters passed in the function as `final`?

Reply:
Good catch. Setting to `final` will potentially allow the Java JIT to apply some optimizations. Regrettably, `final` is not the same as `const` in C/C++ where it will make the object immutable. It will, however, allow the IDE and JIT to error if you try reassigning the object to new memory.

Changes affected.